sdk-workflows/src/main/java/io/dapr/workflows/runtime/WorkflowContext.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.workflows.runtime;
+
+import com.microsoft.durabletask.Task;
+import java.time.Duration;
+
+/**
+ * Context object used by workflow implementations to perform actions such as scheduling activities,
+ * durable timers, waiting for external events, and for getting basic information about the current
+ * workflow instance.
+ */
+public interface WorkflowContext {
+
+  String getName();
+
+  String getInstanceId();
+
+  void complete(Object o);
+
+  Task waitForExternalEvent(String eventName, Duration timeout);
+}
```

Comment:
I assume that over the course of this PR we'll add the other features, like calling activities, timers, and child-workflows?

Reply:
It would be my preference to merge this PR at its current level of functionality to allow for mgmt and authoring APIs to be contributed in parallel. But the minimum feature set required to merge this PR is ultimately something that requires agreement from the maintainers, so feedback certainly welcome.

If we were to expand upon something, I would say the mgmt APIs would be more valuable.  These initial authoring APIs are only implemented for ease of testing.